src/framework/vst/internal/synth/vstaudioclient.cpp

```diff
@@ -0,0 +1,153 @@
+#include "vstaudioclient.h"
+
+#include "audio/synthtypes.h"
+
+using namespace mu;
+using namespace mu::vst;
+
+VstAudioClient::~VstAudioClient()
+{
+    IAudioProcessorPtr processor = m_pluginComponent;
+    if (!processor) {
+        return;
+    }
+
+    processor->setProcessing(false);
+    m_pluginComponent->setActive(false);
+}
+
+void VstAudioClient::init(PluginComponentPtr component)
+{
+    if (!component) {
+        return;
+    }
+
+    m_pluginComponent = component;
+}
+
+bool VstAudioClient::handleEvent(const mu::midi::Event& e)
+{
+    if (m_eventList.addEvent(vstFromMidi(e)) == Steinberg::kResultTrue) {
+        return true;
+    }
+
+    return false;
+}
+
+void VstAudioClient::process(float* output, unsigned int samples)
+{
+    IAudioProcessorPtr processor = m_pluginComponent;

```

Comment:
maybe better to add a method to get a processor

Reply:
There was no reason for that previously since it was the same amount of symbols. But now I'm using static_cast there, so private getter would makes sense